---
layout: post
title: "Unix / Linux Task Session 2"
author: "LÃ©on Kuchenbecker"
categories: task
tags: [task]
---

## Exercise 1

Count the number of sequences (records) in the file `large.fasta`.


## Exercise 2

**The file required for this exercise is `clinvar.vcf.gz`**.

The [VCF Format](https://en.wikipedia.org/wiki/Variant_Call_Format) is a TAB separated format describing genetic variants in comparison to a reference genome. Output the first 7 columns of the first 20 rows of the compressed VCF file `clinvar.vcf.gz`. Exclude the header (lines starting with a `#` character). To solve this exercise, you need to familiarize youself with the tool `cut` (read the manpage!). Your output should look like this:
```
1	1014042	475283	G	A	.	.
1	1014143	183381	C	T	.	.
1	1014217	475278	C	T	.	.
1	1014228	402986	G	A	.	.
1	1014316	161455	C	CG	.	.
1	1014359	161454	G	T	.	.
1	1014451	475281	C	T	.	.
1	1014471	475282	G	C	.	.
1	1020177	474165	C	A	.	.
1	1020183	387476	G	C	.	.
1	1020216	377270	C	G	.	.
1	1020217	128310	G	T	.	.
1	1020221	388958	C	T	.	.
1	1020239	210112	G	C	.	.
1	1022188	263166	A	G	.	.
1	1022225	243036	G	A	.	.
1	1022260	128296	C	T	.	.
1	1022313	243037	A	T	.	.
1	1035307	474141	C	T	.	.
1	1040679	210111	C	T	.	.
```

## Exercise 3

The 5th column of the VCF format describes the alternative allele in contrast to the reference genotype denoted in column 4. List all unique types of mutations and sort them by their frequency in the provided VCF file. Which mutation is the most frequent one?

## Exercise 4

The compressed [FASTQ](https://en.wikipedia.org/wiki/FASTQ_format) file `sequences.fastq.gz` that was downloaded earlier during the course has the read length information stored in every FASTQ ID. Generate an output showing how many times reads of a particular length occur in the file. The output may look like this:
```
      1 length=141
      7 length=143
     99 length=144
   6989 length=146
   8211 length=145
  15328 length=148
  26306 length=147
  79036 length=149
 114023 length=150
```

